tensorflow/core/kernels/fused_batch_norm_op.cc

```diff
@@ -607,14 +623,34 @@ class FusedBatchNormGradOp : public OpKernel {
     OP_REQUIRES_OK(
         context, context->allocate_output(4, TensorShape({}), &placeholder_2));
 
-    functor::FusedBatchNormGrad<Device, T>()(
-        context, y_backprop, x, scale, saved_mean, saved_maybe_inv_var,
-        epsilon_, x_backprop, scale_backprop, offset_backprop, tensor_format_);
+    if (is_training_) {
+      functor::FusedBatchNormGrad<Device, T>()(
+          context, y_backprop, x, scale, saved_mean, saved_maybe_inv_var,
```

Comment:
This is a bit confusing. Rename to saved_mean_or_pop_mean and saved_maybe_inv_var_or_pop_var?

Reply:
Did some rename (on one existing kernel as well) to improve clarity.